tests/lib/screenshot-testing/support/page-renderer.js

```diff
@@ -121,14 +120,19 @@ PageRenderer.prototype._reset = function () {
 PageRenderer.prototype.waitFor = function (selectorOrTimeoutOrFunction) {
     console.log('Using page.waitFor is deprecated, please use one of this instead: waitForSelector, waitForFunction, waitForTimeout');
     if (typeof selectorOrTimeoutOrFunction === 'function') {
-        this.webpage.waitForFunction(selectorOrTimeoutOrFunction)
+        return this.webpage.waitForFunction(selectorOrTimeoutOrFunction)
     } else if (typeof selectorOrTimeoutOrFunction === 'number') {
-        this.webpage.waitForTimeout(selectorOrTimeoutOrFunction)
+        return this.webpage.waitForTimeout(selectorOrTimeoutOrFunction)
     } else if (typeof selectorOrTimeoutOrFunction === 'string') {
-        this.webpage.waitForSelector(selectorOrTimeoutOrFunction)
+        return this.webpage.waitForSelector(selectorOrTimeoutOrFunction)
```

Comment:
waitFor was not working before since it didn't return the promises returned by these functions. It would return immediately instead of waiting.